google/generativeai/types/content_types.py

```diff
@@ -164,16 +163,19 @@ def to_blob(blob: BlobType) -> glm.Blob:
     elif isinstance(blob, IMAGE_TYPES):
         return image_to_blob(blob)
     else:
-        if isinstance(blob, Mapping):
+        if "parts" not in blob:
```

Comment:
There's something wrong here. 

I don't see how it can be a Mapping at this point: Mapping is handled first, and _convert_dict never returns a Mapping.

I think we only need the else clause here. 

Maybe put that change in a separate PR, so this is only docs and  error messages.